### PR TITLE
RMSW-77: AnalogSensor.c - HAL Code Update

### PR DESCRIPTION
**Pull request by tiffcmw**
Updated analog sensor code (AnalogSensor.c) for GPIO and channel mappings using STM32 HAL.

### CODE REVIEW

File: Core/Src/Sensors/AnalogSensor.c [hunk 1]
```diff
@@ -1,7 +1,187 @@
 #include "../../Inc/Sensors/AnalogSensor.h"
+#include <stdio.h>
+
+void initAnalogSensor(AnalogSensor* sensor, const char* name, int hz, int channel) {
+    initSensor(&sensor->base, name, hz);
+    sensor->channel = channel;
+
+    if(sensor->channel == 0) {
+    	// HAL code to bind to PA0
+    	__HAL_RCC_GPIOA_CLK_ENABLE();
```
Comment (Arct1cPharaoh) [inline Core/Src/Sensors/AnalogSensor.c:10]: There are statements like this that could be applied in ranges to make it make it more readable. ie. 
```c
if (sensor->channel <= 7) {
        __HAL_RCC_GPIOA_CLK_ENABLE();
        HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);
...
```

File: Core/Src/Sensors/AnalogSensor.c [hunk 2]
```diff
@@ -1,7 +1,187 @@
...
+    	GPIO_InitTypeDef GPIOAout_Struct;
+		GPIOJout_Struct.Pin = GPIO_PIN_0;
+		GPIOJout_Struct.Mode = GPIO_MODE_OUTPUT_PP;
```
Comment (Arct1cPharaoh) [inline Core/Src/Sensors/AnalogSensor.c:13]: Since this statement and other statements are in every case you could move them out of the conditionals.

**Review by Arct1cPharaoh — COMMENTED** [review]
A few improvements to possibly reduce number of lines.

File: Core/Src/Sensors/AnalogSensor.c [hunk 3]
```diff
@@ -1,7 +1,187 @@
 #include "../../Inc/Sensors/AnalogSensor.h"
+#include <stdio.h>
+
+void initAnalogSensor(AnalogSensor* sensor, const char* name, int hz, int channel) {
+    initSensor(&sensor->base, name, hz);
+    sensor->channel = channel;
+
+    if(sensor->channel == 0) {
```
[the remark this answers is not in the document]
Comment (tiffcmw) [reply Core/Src/Sensors/AnalogSensor.c]: Thank you so much for your comments! Just corrected it.

File: Core/Src/Sensors/AnalogSensor.c [hunk 4]
```diff
@@ -1,7 +1,124 @@
 #include "../../Inc/Sensors/AnalogSensor.h"
+#include <stdio.h>
 
-void initAnalogSensor(AnalogSensor* analogsensor, const char* name, int hz, int channel) {
-    initSensor(&analogsensor->sensor, name, hz, ANALOG);
-    analogsensor->channel = channel;
+void initAnalogSensor(AnalogSensor* sensor, const char* name, int hz, int channel) {
```
Comment (AmarMaksumic) [inline Core/Src/Sensors/AnalogSensor.c:4]:
```suggestion
// Add function doc here! Look at any function in VCU C-Code for an example~
void initAnalogSensor(AnalogSensor* sensor, const char* name, int hz, int channel) {
```

**Review by AmarMaksumic — APPROVED** [review]
Great first PR!

**Pull request merged**